### PR TITLE
New patch release of Golang: v1.21.6

### PR DESCRIPTION
**Pull request by jngo2**
Update EKS Go Patch Version: v1.21.6-7
SPEC FILE STILL NEEDS THE '%changelog' UPDATED
PLEASE UPDATE WITH THE FOLLOWING FORMAT
```
* Wed Sep 06 2023 Cameron Rozean <rcrozean@amazon.com> - 1.20.8-1
- Bump tracking patch version to 1.20.8 from 1.20.7
```

### CODE REVIEW

File: projects/golang/go/1.21/rpmbuild/SPECS/golang.spec
Comment (zafs23) [inline projects/golang/go/1.21/rpmbuild/SPECS/golang.spec:1]: @jngo2 please add readme changes as well. Follow https://github.com/aws/eks-distro-build-tooling/pull/1272 changes

Comment (rcrozean) [reply projects/golang/go/1.21/rpmbuild/SPECS/golang.spec:1]: Read me changes come later in the automation.

**Pull request merged**